client/lib/numalib/detect_linux_test.go

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+//go:build linux
+
+package numalib
+
+import (
+	"os"
+	"testing"
+
+	"github.com/hashicorp/nomad/client/lib/idset"
+	"github.com/hashicorp/nomad/client/lib/numalib/hw"
+	"github.com/shoenig/test/must"
+)
+
+// badSysData are example values from sysfs on unsupported platforms, e.g.,
+// containers, virtualization guests
```

Comment:
There's a good bit here, so it'd be helpful if there were a comment alongside the lines that are different indicating what's unexpected.

Reply:
I added extra comments